biggraphite/metadata_cache.py

```diff
@@ -402,6 +411,23 @@ def __split_payload(self, payload):
             return []
         return payload.split(self._METRIC_SEPARATOR, 2)
 
+    def clean(self):
+        """Remove all expired metrics.
+
+        Note: This will also remove metrics without a timestamp or
+              whose timestamp value is not a digit.
+        """
+        with self.__env.begin(self.__metric_to_metadata_db, write=True) as txn:
+            cursor = txn.cursor()
+            for key, value in cursor:
+                timestamp_str = self.__split_payload(value)[-1]
```

Comment:
That would fail if there is no timestamp (e.g. previous version of the database)